projects/npm-tools/packages/npm-scripts/src/utils/getJestModuleNameMapper.js

```diff
@@ -33,10 +29,9 @@ const IGNORE_GLOBS = ['node_modules/**'];
  *
  * We create such mappings by:
  *
- *    1. Reading the Yarn workspace globs defined in the top-level
- *       "modules/package.json".
- *    2. Expanding those globs to identify possible projects.
- *    3. Selecting only projects which have a "package.json" with a "main"
+ *    1. Iterating over projects identified by the Yarn workspace globs
+ *       defined in the top-level "modules/package.json".
+ *    2. Selecting only projects which have a "package.json" with a "main"
```

Comment:
What's the reason for this? Not that I'm against, just wondering...

Reply:
The need for a valid `main` value is explained a bit higher up in the comment. It's because we need to generate a pair of mappings, and one of them references the `main` file:

```json
{
    "^frontend-js-web$": "<rootDir>../../frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.es.js",
    "^frontend-js-web/(.*)": "<rootDir>../../frontend-js/frontend-js-web/src/main/resources/META-INF/resources/$1"
}
```

That's what makes `import {x} from 'frontend-js-web'` work in the context of Jest, because Jest can figure out that it actually means `<rootDir>../../frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.es.js`.

Reply:
What would happen if you had a module that didn't export anything and wasn't imported by anyone? Would it need to create a bogus `main` value in order to be able to use TS internally?

Reply:
I think all modules should have a `main`, even if it doesn't export anything (rare, but modules that exist solely for their side-effects: all the parts of the system still need to now _what_ to run, and that's what `main` is for).